commit 5b8054c6b9fdbdd5cfe55e67057ca0ad4907c4d8
Author: Ryan Tandy <rtandy@sd63.bc.ca>
Date:   Wed Sep 24 10:10:29 2014 -0700

    Update gnome-session commands for GNOME 3
    
    gnome-session-quit has no --reboot option, and --power-off doesn't
    respect --no-prompt, so just make the appropriate dbus calls directly.

diff --git a/ica/src/LocalSystemIca.cpp b/ica/src/LocalSystemIca.cpp
index 732bdba..cf84236 100644
--- a/ica/src/LocalSystemIca.cpp
+++ b/ica/src/LocalSystemIca.cpp
@@ -141,8 +141,7 @@ void powerDown( void )
 	}
 	else
 	{
-		QProcess::startDetached( "gdm-signal -h" ); // Gnome shutdown
-		QProcess::startDetached( "gnome-session-save --silent --kill" ); // Gnome logout
+		QProcess::startDetached( "dbus-send --session --dest=org.gnome.SessionManager --type=method_call /org/gnome/SessionManager org.gnome.SessionManager.RequestShutdown" ); // Gnome shutdown
 		QProcess::startDetached( "dcop ksmserver ksmserver logout 0 2 0" ); // KDE shutdown
 		QProcess::startDetached( "qdbus org.kde.ksmserver /KSMServer logout 0 2 0" ); // KDE4 shutdown
 	}
@@ -157,7 +156,7 @@ void logoutUser( void )
 #ifdef ITALC_BUILD_WIN32
 	ExitWindowsEx( EWX_LOGOFF | SHUTDOWN_FLAGS, SHUTDOWN_REASON );
 #else
-	QProcess::startDetached( "gnome-session-save --silent --kill" ); // Gnome logout
+	QProcess::startDetached( "dbus-send --session --dest=org.gnome.SessionManager --type=method_call /org/gnome/SessionManager org.gnome.SessionManager.Logout uint32:2" ); // Gnome logout, 2 = forced mode (don't wait for unresponsive processes)
 	QProcess::startDetached( "dcop ksmserver ksmserver logout 0 0 0" ); // KDE logout
 	QProcess::startDetached( "qdbus org.kde.ksmserver /KSMServer logout 0 0 0" ); // KDE4 logout
 #endif
@@ -178,8 +177,7 @@ void reboot( void )
 	}
 	else
 	{
-		QProcess::startDetached( "gdm-signal -r" ); // Gnome reboot
-		QProcess::startDetached( "gnome-session-save --silent --kill" ); // Gnome logout
+		QProcess::startDetached( "dbus-send --session --dest=org.gnome.SessionManager --type=method_call /org/gnome/SessionManager org.gnome.SessionManager.RequestReboot" ); // Gnome reboot
 		QProcess::startDetached( "dcop ksmserver ksmserver logout 0 1 0" ); // KDE reboot
 		QProcess::startDetached( "qdbus org.kde.ksmserver /KSMServer logout 0 1 0" ); // KDE4 reboot
 	}
